### PR TITLE
[#349] Allow enkelvoudiginformatieobjecten to be destroyed with the n…

### PR DESCRIPTION
**Pull request by joeribekker**
…ew scope: scopes.documenten.verwijderen

### CODE REVIEW

File: src/drc/tests/test_userstory_349.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        ObjectInformatieObjectFactory.create(informatieobject=informatieobject, is_zaak=True)
+
+        informatieobject_delete_url = get_operation_url('enkelvoudiginformatieobject_delete', uuid=informatieobject.uuid)
+
+        response = self.client.delete(informatieobject_delete_url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT, response.data)
+
+        self.assertEqual(EnkelvoudigInformatieObject.objects.all().count(), 0)
+
+        self.assertEqual(Gebruiksrechten.objects.all().count(), 0)
```
Comment (sergei-maertens) [inline src/drc/tests/test_userstory_349.py]:
```suggestion
        self.assertFalse(Gebruiksrechten.objects.all().exists())
```

File: src/drc/tests/test_userstory_349.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        self.assertEqual(ObjectInformatieObject.objects.all().count(), 0)
```
Comment (sergei-maertens) [inline src/drc/tests/test_userstory_349.py]:
```suggestion
        self.assertFalse(ObjectInformatieObject.objects.all().exists())
```

File: src/drc/api/tests/test_objectinformatieobject.py [hunk 3]
```diff
@@ -37,9 +37,15 @@ class ObjectInformatieObjectAPITests(APITestCase):
...
-        patcher = patch('drc.sync.signals.sync_create')
-        self.mocked_sync_create = patcher.start()
-        self.addCleanup(patcher.stop)
+        super().setUp()
+
+        patcher_sync_create = patch('drc.sync.signals.sync_create')
+        self.mocked_sync_create = patcher_sync_create.start()
+        self.addCleanup(patcher_sync_create.stop)
+
+        patcher_sync_delete = patch('drc.sync.signals.sync_delete')
```
[the remark this answers is not in the document]
Comment (joeribekker) [reply src/drc/api/tests/test_objectinformatieobject.py]: It is: https://github.com/VNG-Realisatie/gemma-documentregistratiecomponent/pull/37/files#diff-1a6629e4b4b0491a5bc4c92f2280699a

File: src/drc/api/viewsets.py [hunk 4]
```diff
@@ -57,16 +60,23 @@ class EnkelvoudigInformatieObjectViewSet(mixins.CreateModelMixin,
...
+    Verwijdert een ENKELVOUDIG INFORMATIEOBJECT, samen met alle gerelateerde resources binnen deze API.
     """
     queryset = EnkelvoudigInformatieObject.objects.all()
     serializer_class = EnkelvoudigInformatieObjectSerializer
     lookup_field = 'uuid'
-
+    permission_classes = (ActionScopesRequired, )
+    required_scopes = {
+        'delete': SCOPE_DOCUMENTEN_ALLES_VERWIJDEREN,
+    }
```
[the remark this answers is not in the document]
Comment (joeribekker) [reply src/drc/api/viewsets.py]: Hmm, i checked the` ActionScopeRequired` for this, and noticed it handles undefined as: no permission needed, but could add a test for this.

File: src/drc/api/tests/test_objectinformatieobject.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/drc/api/tests/test_objectinformatieobject.py]: yep, I forgot to delete my comment when I came across that. I like that implementation!

File: src/drc/api/viewsets.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/drc/api/viewsets.py]: yeah I double checked, you're right. I'm assuming there already is a test for this, but feel free to add it if there isn't

**Comment by joeribekker** [comment]
Resolved all issues.

File: src/drc/api/tests/test_objectinformatieobject.py [hunk 5]
```diff
@@ -320,3 +326,28 @@ def test_sync_create_fails(self):
...
+        self.assertEqual(self.mocked_sync_delete.call_count, 0)
+
+        response = self.client.delete(oio_url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT, response.data)
+
+        self.assertEqual(self.mocked_sync_delete.call_count, 1)
+
+        # Relation is gone, document still exists.
+        self.assertEqual(ObjectInformatieObject.objects.count(), 0)
+        self.assertEqual(EnkelvoudigInformatieObject.objects.count(), 1)
```
[the remark this answers is not in the document]
Comment (joeribekker) [reply src/drc/api/tests/test_objectinformatieobject.py]: @sergei-maertens Nice, It should be assertTrue ;-)

[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/drc/api/tests/test_objectinformatieobject.py]: yikes, force of habit :D

**Pull request merged**